### ISSUE TITLE
OAR restful API missing dependency on Debian: libyaml-perl

### ISSUE THREAD

**Issue by mickours** [oar-team/oar#10]
The package ``libyaml-perl`` is mark as recommended for the oar-restful-api debian package but the default behavior is to show content as YAML /oarapi/jobs.html for example:
```
Status: 400 Content-Type: text/html; charset=ISO-8859-1
YAML not enabled

YAML perl module not loaded!
```
It should be required (better I think) or the default content should be json files (less readable)

**Comment by bzizou** [comment]
You need at least one of the yaml or the json perl module, but none is mandatory. If you only use json output, the API is fully useable without the yaml package. In the past, I think that there was an "or" dependency (libjson-perl | libyaml-perl ). I don't know why it has been converted into a "json by default, yaml recommended"...

**Comment by npf** [comment]
git blame debian/control says it's you :)

4d97b204 (Bruno Bzeznik        2009-09-24 13:18:32 +0000 136) Recommends: libxml-simple-perl, libyaml-perl

See gitk debian/control

Question, should we change it ? I think no.
From my viewpoint, json is good as the default for a rest api (but I'm not an expert) so if @mickours is correct, I would choose the second option: always return json by default, and no require Yaml, but just suggest it.

**Comment by npf** [comment]
@SalemHarrache @pmorillon , what do you thing ?

**Comment by bzizou** [comment]
OK, my fault, the json dependency is there since 2009 and it's my decision :-)
For better simplicity, I suggest that the API requires both libyaml-perl and libjson-perl. 

But here is the reason why it's not the case: the fact is that there's no default output. You always have to choose between json, yaml or html (by setting the header or by giving a .json, .yaml or .html extension). Here, you asked for an html output (using the .html extension) and this requires yaml as the html output uses a yaml formatting for the human readability. But the html output is not the main purpose of this API that is generally used with yaml or json encoding. The message "YAML perl module not loaded" suggests that for the html output, you need to install the yaml perl module. So, libyaml-perl is not strictly required, but only suggested in the case you need yaml or html output.

Anyway, it may be simpler for everyone if the oar-restful-api package requires both libyaml-perl and libjson-perl as it's not a very big dependency, what do you think?

**Comment by SalemHarrache** [comment]
About the packaging, I agree with Bruno, we should keep both dependences *for now* : it is lightweight and yaml can be used out of the box.

However, for operating the API, I think the best solution is to give priority to json.

We only keep YAML because it requires minimum effort (converting a hash into yaml is trivial). The API is not specially designed for humans (readability).

We don't need any other format (html?) in my opinion.

**Comment by npf** [comment]
What about libxml-simple-perl then ? Recommand only or require as well ?

**Comment by bzizou** [comment]
@npf : you can completely remove libxml-simple-perl from recommandation, there's no more XML support

**Comment by npf** [comment]
fixed in Debian package 2.5.4-2
